CMakeLists.txt

```
@@ -9,12 +9,39 @@ Emscripten.cmake"
     )
 endif()
 
+# Extract the project name & version from package.json
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/package.json")
+    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/package.json" PACKAGE_JSON_CONTENT)
+else()
+    message(FATAL_ERROR "`package.json` not found in ${CMAKE_CURRENT_SOURCE_DIR}")
+endif()
+if(${PACKAGE_JSON_CONTENT} MATCHES "\"name\":[ ]*\"([^\"]+)\"")
```

```suggestion
if("${PACKAGE_JSON_CONTENT}" MATCHES "\"name\":[ ]*\"([^\"]+)\"")
```
To avoid [double-variable expansion](https://cmake.org/cmake/help/latest/command/if.html#variable-expansion).